### PR TITLE
fix(bug): out of memory

### PR DESCRIPTION
**Pull request by alebinson**
Think the cause of the bug was a few things:
1. UI freeze -> long loop when Bbox is big or Max zoom is big as well
Resolution: asynchronous is added (see code, nice trick with resolved promise)
2. Big amount of memory is allocated during tiles list calculation(garbage collection issue)
Resolution: added function that returns only the estimated tiles count(number)
3. Debounce  added to avoid unneeded expansive calculations 
4. Theme is always LIGHT (TODO when dark theme will be tuned)

### CODE REVIEW

File: src/common/helpers/estimated-tile-list.ts [hunk 1]
```diff
@@ -87,11 +91,72 @@ const getTiles = (polygon: Polygon, minZoom: number, maxZoom: number): Array<str
...
   });
   return res;
-}
+};
+
+const getTilesCount = (polygon: Polygon, minZoom: number, maxZoom: number): number => {
+  const onlyCorners = false;
+  const zooms = Array.apply(null, Array(maxZoom - minZoom + 1)).map( (x, i) => (minZoom + i) );
+
+  const oneMoment = () => new Promise(resolve => setTimeout(resolve));
```
Comment (asafMasa) [inline src/common/helpers/estimated-tile-list.ts:109]: defined twice -. get out from function scope?

File: src/common/helpers/estimated-tile-list.ts [hunk 2]
```diff
@@ -87,11 +91,72 @@ const getTiles = (polygon: Polygon, minZoom: number, maxZoom: number): Array<str
...
+  const iterationsPerChunk = 10000;
```
Comment (asafMasa) [inline src/common/helpers/estimated-tile-list.ts:110]: defined twice -. get out from function scope?

File: src/common/helpers/estimated-tile-list.ts [hunk 3]
```diff
@@ -87,11 +91,72 @@ const getTiles = (polygon: Polygon, minZoom: number, maxZoom: number): Array<str
...
+        }
       }
     }
     //   })
   });
   return res;
-}
+};
+
+const getTilesCount = (polygon: Polygon, minZoom: number, maxZoom: number): number => {
```
Comment (asafMasa) [inline src/common/helpers/estimated-tile-list.ts:105]: consider reuse of those 2 functions parts instead of duplicating them

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/common/helpers/estimated-tile-list.ts (hunk 3 above)
Comment (alebinson) [reply src/common/helpers/estimated-tile-list.ts:105]: That's the problem was, should be 2 of them, garbage collection just blows when generating tile images names.
Also for performance reason should be 2 of them

File: src/common/helpers/estimated-tile-list.ts (hunk 1 above)
Comment (alebinson) [reply src/common/helpers/estimated-tile-list.ts]: Nice "eagle eye" :)
Done

File: src/common/helpers/estimated-tile-list.ts (hunk 2 above)
Comment (alebinson) [reply src/common/helpers/estimated-tile-list.ts]: Nice "eagle eye" :)
Done

**Review by asafMasa — APPROVED** [review]

**Pull request merged**